lib/util/package-rules.ts

```diff
@@ -155,7 +155,10 @@ function matchesRule(
     matchPackagePrefixes.length
   ) {
     if (!depName) {
-      return false;
+      return (
+        is.undefined(packageRule.matchPackagePatterns) ||
+        packageRule.matchPackagePatterns.length !== 0
```

Comment:
Is it your intention that in the case of e.g. matchPackageNames that it does not return false here?